main_app/models.py

```diff
@@ -81,10 +81,10 @@ def get_item_count(self):
     return count
   get_item_count.short_description = 'Number of items'
 
-  # class Meta:
-  #   constraints = [
-  #     models.UniqueConstraint(fields=['user'], condition=Q(status=0), name='unique_current_cart')
-  #   ]
+  class Meta:
+    constraints = [
+      models.UniqueConstraint(fields=['user','status'], condition=Q(status=1), name='unique_current_cart')
```

Comment:
Most important line. Pass in the two fields that you want to be unique together and a conditions so that it's only unique for `CURRENT` carts

Reply:
slick!